metagraph/src/cli/query.cpp

```diff
@@ -404,6 +403,14 @@ void add_nodes_with_suffix_matches(const DBGSuccinct &full_dbg,
         call_suffix_match_sequences(full_dbg, contig, path,
                                     callback, sub_k, max_num_nodes_per_suffix);
 
+        if (check_reverse_complement && !full_dbg.is_canonical_mode()) {
+            std::string rev_contig = contig;
+            reverse_complement(rev_contig);
+            call_suffix_match_sequences(full_dbg, rev_contig,
+                                        map_sequence_to_nodes(full_dbg, rev_contig),
+                                        callback, sub_k, max_num_nodes_per_suffix);
+        }
+
```

Comment:
Maybe we should do this even if full_dbg is canonical? Because suffixes in dual nodes are prefixes in the forward ones.

Reply:
It should still be fine in most cases, but we should still do this for all graph types, since it will fail if the suffix match is to a dummy source k-mer.

Reply:
Hm, not sure I got it? Do you mean we should fix something here or you think it's good?

Reply:
I think that it should be fixed, as in, we should check the suffixes for dual nodes in all cases when the batch graph is canonical

Reply:
I just wanted to point out that it's a corner case